sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -0,0 +1,174 @@
+    DateTime operator+(IntervalType value) const { return DateTime(m_interval + value); }
+
+    /// Compare two instances of @DateTime for equality.
+    bool operator==(DateTime dt) const { return m_interval == dt.m_interval; }
+
+    /// Compare two instances of @DateTime for inequality.
+    bool operator!=(const DateTime& dt) const { return !(*this == dt); }
+
+    /// Compare the chronological order of two @DateTime instances.
+    bool operator>(const DateTime& dt) const { return this->m_interval > dt.m_interval; }
+
+    /// Compare the chronological order of two @DateTime instances.
+    bool operator<(const DateTime& dt) const { return this->m_interval < dt.m_interval; }
+
+    /// Compare the chronological order of two @DateTime instances.
+    bool operator>=(const DateTime& dt) const { return this->m_interval >= dt.m_interval; }
+
+    /// Compare the chronological order of two @DateTime instances.
+    bool operator<=(const DateTime& dt) const { return this->m_interval <= dt.m_interval; }
+
+    /// Create an interval from milliseconds.
+    static IntervalType FromMilliseconds(unsigned int milliseconds)
+    {
+      return milliseconds * TicksPerMillisecond;
+    }
+
+    /// Create an interval from seconds.
+    static IntervalType FromSeconds(unsigned int seconds) { return seconds * TicksPerSecond; }
+
+    /// Create an interval from minutes.
+    static IntervalType FromMinutes(unsigned int minutes) { return minutes * TicksPerMinute; }
+
+    /// Create an interval from hours.
+    static IntervalType FromHours(unsigned int hours) { return hours * TicksPerHour; }
+
+    /// Create an interval from days.
+    static IntervalType FromDays(unsigned int days) { return days * TicksPerDay; }
+
+    /// Checks whether this instance of @DateTime is initialized.
+    bool IsInitialized() const { return m_interval != 0; }
+
+  private:
+    friend IntervalType operator-(DateTime t1, DateTime t2);
+
+    static constexpr IntervalType TicksPerMillisecond = static_cast<IntervalType>(10000);
+    static constexpr IntervalType TicksPerSecond = 1000 * TicksPerMillisecond;
+    static constexpr IntervalType TicksPerMinute = 60 * TicksPerSecond;
+    static constexpr IntervalType TicksPerHour = 60 * TicksPerMinute;
+    static constexpr IntervalType TicksPerDay = 24 * TicksPerHour;
+
+    // Private constructor. Use static methods to create an instance.
+    DateTime(IntervalType interval) : m_interval(interval) {}
+
+    // Storing as hundreds of nanoseconds 10e-7, i.e. 1 here equals 100ns.
+    IntervalType m_interval;
+  };
+
+  inline DateTime::IntervalType operator-(DateTime t1, DateTime t2)
+  {
+    auto diff = (t1.m_interval - t2.m_interval);
+
+    // Round it down to seconds
+    diff /= DateTime::TicksPerSecond;
+
+    return static_cast<DateTime::IntervalType>(diff);
+  }
+
+  /**
+   * @brief An exception that gets thrown when @DateTime error occurs.
+   */
+  class DateTimeException : public std::runtime_error {
```

Comment:
once https://github.com/Azure/azure-sdk-for-cpp/pull/717 is merged, 
do we want DateTimeException to derive from RequestFailedException as all other excpetions?

Reply:
I thought, RequestFailedException is for HTTP requests?